azure-pipelines.yml

```diff
@@ -90,74 +119,125 @@ stages:
 
   - stage: Genesis
     displayName: 'Genesis'
-    condition: ne('${{ parameters.Action }}', 'Destroy')
     jobs:
       - job: Genesis
         steps:
-          - template: pipeline-steps/genesis.yaml
+          - template: steps/terraform.yaml@cnp-azuredevops-libraries
             parameters:
+              overrideAction: ${{ parameters.action }}
               environment: ${{ parameters.env }}
+              component: 'genesis'
               location: ${{ parameters.location }}
-              stack: 'genesis'
-              project: $(project)
-              tfversion: $(tfversion)
-              tfInitSub: ${{ variables.tfInitSub }}
-              builtFrom: $(Build.Repository.Name)
+              serviceConnection: $(serviceConnection)
+              terraformInitSubscription: ${{ variables.tfInitSub }}
               product: ${{ variables.product }}
+              terraformVarFile: NULL
+              initCommandOptions: >
+                -var subscription_id=$(ARM_SUBSCRIPTION_ID)
+                -backend-config resource_group_name=$(backendAzureRmResourceGroupName)
+              planCommandOptions: >
+                -var subscription_id=$(ARM_SUBSCRIPTION_ID)
 
   - stage: Network
     displayName: 'Network'
-    condition: ne('${{ parameters.Action }}', 'Destroy')
     jobs:
       - job: NetworkRg
         steps:
-          - template: pipeline-steps/deploy-service.yaml
+          - template: steps/set-build-repo-suffix-env-var.yaml@cnp-azuredevops-libraries
+          - template: steps/terraform.yaml@cnp-azuredevops-libraries
             parameters:
+              overrideAction: ${{ parameters.action }}
               environment: ${{ parameters.env }}
+              component: 'network-rg'
               location: ${{ parameters.location }}
-              stack: 'network-rg'
-              project: $(project)
-              tfversion: $(tfversion)
-              tfInitSub: ${{ variables.tfInitSub }}
-              builtFrom: $(Build.Repository.Name)
+              serviceConnection: $(serviceConnection)
+              terraformInitSubscription: ${{ variables.tfInitSub }}
               product: ${{ variables.product }}
+              terraformVarFile: "$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/network-rg/${{ parameters.env }}.tfvars"
+              initCommandOptions: >
+                -var subscription_id=$(ARM_SUBSCRIPTION_ID)
+                -backend-config resource_group_name=$(backendAzureRmResourceGroupName)
+              planCommandOptions: >
+                -var control_vault=$(controlKeyVault)
+                -var project=$(project)
+
       - job: DeployInfrastructure
         dependsOn: NetworkRg
         steps:
-          - template: pipeline-steps/deploy-network.yaml
+          - template: steps/set-build-repo-suffix-env-var.yaml@cnp-azuredevops-libraries
+
+          - task: AzureCLI@2
+            displayName: Determine Key Vault and Storage names
```

Comment:
why did you need to copy this, you should just be able to use https://github.com/hmcts/cnp-azuredevops-libraries/blob/master/steps/keyvault-read.yaml